### PR TITLE
ref(onboarding-docs): Cleanup SdkDocumentation component

### PR DESCRIPTION
**Pull request by ArthurKnaus**
Remove fallback for exported Docs component (all docs are migrated to exporting objects)
Extract, refactor and test `getPlatformPath`.

### CODE REVIEW

File: static/app/utils/gettingStartedDocs/getPlatformPath.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import type {PlatformIntegration} from 'sentry/types/project';
+
+export function getPlatformPath(platform: PlatformIntegration) {
+  // TODO: This will be removed once we no longer rely on sentry-docs to load platform icons
```
Comment (ArthurKnaus) [inline static/app/utils/gettingStartedDocs/getPlatformPath.ts:4]: @priscilawebdev you added this todo, right?
Let's sync at one point if we are ready to get rid of this logic.

File: static/app/components/onboarding/gettingStartedDoc/sdkDocumentation.tsx [hunk 2]
```diff
@@ -126,22 +72,6 @@ export function SdkDocumentation({
 
   const {default: docs} = module;
 
-  if (isFunctionalComponent(docs)) {
```
Comment (priscilawebdev) [inline static/app/components/onboarding/gettingStartedDoc/sdkDocumentation.tsx:129]: @ArthurKnaus why did we introduce this logic and why are we removing it?

Comment (ArthurKnaus) [reply static/app/components/onboarding/gettingStartedDoc/sdkDocumentation.tsx:129]: In a previous version the docs files exported a component that rendered the onboarding docs.
We gradually migrated them to return the docs as an object to decouple the content from how they are rendered in different scenarios.

This migration is complete so we can remove the old logic branch (that handles the component version).

File: static/app/utils/gettingStartedDocs/getPlatformPath.spec.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+      language: 'python',
+      link: 'link',
+      name: 'name',
+    };
+
+    expect(getPlatformPath(platform)).toEqual('python/python');
+  });
+
+  it('handles special cases', () => {
+    function getFrameworklatformWithId(id: PlatformKey): PlatformIntegration {
```
Comment (priscilawebdev) [inline static/app/utils/gettingStartedDocs/getPlatformPath.spec.ts:30]:
```suggestion
    function getFrameworkPlatformWithId(id: PlatformKey): PlatformIntegration {
```

**Review by priscilawebdev — APPROVED** [review]
apart from the typo it looks good to me

File: static/app/utils/gettingStartedDocs/getPlatformPath.ts (hunk 1 above)
Comment (priscilawebdev) [reply static/app/utils/gettingStartedDocs/getPlatformPath.ts:4]: @ArthurKnaus I have worked on it in the PR https://github.com/getsentry/sentry/pull/72764

**Pull request merged**